Fix warning in core_class_hierarchy.hpp with clang / xcode7.2

The Travis bots reports a warning (xcode7.2 travis image) in core_class_hierarchy.hpp that needs to be fixed.

>/Users/travis/build/gccxml/pygccxml/unittests/data/core_class_hierarchy.hpp:30:44: warning: direct base 'core::class_hierarchy::base_t' is inaccessible due to ambiguity:
    class core::class_hierarchy::multi_derived_t -> class core::class_hierarchy::derived_private_t -> class core::class_hierarchy::base_t
    class core::class_hierarchy::multi_derived_t -> class core::class_hierarchy::base_t [-Winaccessible-base]
class multi_derived_t : derived_private_t, protected base_t, private other_base_t{
                                           ^~~~~~~~~~~~~~~~
1 warning generated.

--- a/unittests/data/core_class_hierarchy.hpp
+++ b/unittests/data/core_class_hierarchy.hpp
@@ -6,8 +6,6 @@
 #ifndef __core_class_hierarchy_hpp__
 #define __core_class_hierarchy_hpp__
 
-//TODO("To add virtual inheritance case");
-
 namespace core{ namespace class_hierarchy{
 
 class base_t{
@@ -24,10 +22,10 @@ class derived_public_t : public base_t{
 class derived_protected_t : protected base_t{
 };
 
-class derived_private_t : private base_t{
+class derived_private_t : private virtual base_t{
 };
 
-class multi_derived_t : derived_private_t, protected base_t, private other_base_t{
+class multi_derived_t : derived_private_t, protected virtual base_t, private other_base_t{
 };
 
 } }

